test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -6037,8 +6037,14 @@ def submit(self, obj, script=None, extend=None, submit_dir=None):
             if (not self._is_local) or \
                     (not self.du.is_localhost(user_host)):
                 for k, v in obj.custom_attrs.items():
-                    if isinstance(v, str) and (" " in v):
-                        v = "'" + v + "'"
+                    if isinstance(v, str):
+                        if " " in v:
+                            v = "'%s'" % v
+                        if any((c in v) for c in set(',\'"')):
+                            if '"' in v:
+                                v = "\\'%s\\'" % v
```

Comment:
why do we need to add `\` here?
Also, I think we should move this fix to `BatchUtils.convert_to_cli` (see https://github.com/PBSPro/pbspro/blob/master/test/fw/ptl/lib/pbs_testlib.py#L2405)

Reply:
I have moved this part of code to convert_to_cli function.
We need \ to escape single quote in set of characters to match.